src/NzbDrone.Core/Movies/MovieRepository.cs

```diff
@@ -32,195 +30,213 @@ public interface IMovieRepository : IBasicRepository<Movie>
+                .Select(select)
+                .Join("Profiles ON Profiles.Id = Movies.ProfileId")
+                .LeftJoin("AlternativeTitles ON AlternativeTitles.MovieId = Movies.Id")
+                .LeftJoin("MovieFiles ON MovieFiles.MovieId = Movies.Id");
         }
 
-        public Movie FindByTitle(string cleanTitle)
+        private Movie Map(Dictionary<int, Movie> dict, Movie movie, Profile profile, AlternativeTitle altTitle, MovieFile movieFile)
         {
-            return FindByTitle(cleanTitle, null);
+            Movie movieEntry;
+
+            if (!dict.TryGetValue(movie.Id, out movieEntry))
+            {
+                movieEntry = movie;
+                movieEntry.Profile = profile;
+                movieEntry.MovieFile = movieFile;
+                dict.Add(movieEntry.Id, movieEntry);
+            }
+
+            if (altTitle != null)
+            {
+                movieEntry.AlternativeTitles.Add(altTitle);
+            }
+
+            return movieEntry;
         }
 
-        public Movie FindByTitle(string cleanTitle, int year)
+        protected override Movie SelectOne(SqlBuilder.Template sql)
         {
-            return FindByTitle(cleanTitle, year as int?);
+            return SelectMany(sql).FirstOrDefault();
         }
 
-        public Movie FindByImdbId(string imdbid)
+        protected override IEnumerable<Movie> SelectMany(SqlBuilder.Template sql)
         {
-            var imdbIdWithPrefix = Parser.Parser.NormalizeImdbId(imdbid);
-            return Query.Where(s => s.ImdbId == imdbIdWithPrefix).SingleOrDefault();
+            // Console.WriteLine($"Sql: {sql.RawSql}, Params: {((DynamicParameters)sql.Parameters).ToJsonString()}");
+            var movieDictionary = new Dictionary<int, Movie>();
+            return DataMapper.Query<Movie, Profile, AlternativeTitle, MovieFile, Movie>(
+                sql.RawSql,
+                (movie, profile, altTitle, file) => Map(movieDictionary, movie, profile, altTitle, file),
+                sql.Parameters)
+                .ToList();
         }
 
-        public List<Movie> GetMoviesByFileId(int fileId)
+        // This doesn't populate profiles - deserializing repeatedly is v slow
+        public override IEnumerable<Movie> All()
         {
-            return Query.Where(m => m.MovieFileId == fileId).ToList();
+            var noProfileTemplate = $"SELECT /**select**/ FROM {_table} /**leftjoin**/ /**where**/ /**orderby**/";
+            var sql = Builder().AddTemplate(noProfileTemplate);
+
+            var movieDictionary = new Dictionary<int, Movie>();
+            DataMapper.Query<Movie, AlternativeTitle, MovieFile, Movie>(
+                sql.RawSql,
+                (movie, altTitle, file) => Map(movieDictionary, movie, null, altTitle, file),
+                sql.Parameters);
+
+            // Doing the join on profile repo is slow - deserialize the same thing too many times
+            var profiles = _profileRepository.All();
```

Comment:
I wonder if we're better of having services just asking the ProfileService when needed rather than always including the profile instances.

Reply:
This is potentially a radarrism, as I mentioned in the discord, to allow running the upgradeable spec on the API all endpoint. In sonarr your suggested approach might make more sense.